mne/viz/_3d.py

```diff
@@ -353,12 +353,56 @@ def plot_trans(info, trans_fname='auto', subject=None, subjects_dir=None,
     return fig
 
 
+def _percent_to_control_points(limits, stc_data, colormap):
+    """Private helper function to convert percentiles to control points.
+
+    Note: If using 'mne_analyze', generate cmap control points for a directly
+    mirrored cmap for simplicity. I.e., no normalization to account for a
+    2-tailed mne_analyze cmap is computed.
+
+    Parameters
+    ----------
+    limits : str | 3-tuple | dict
+        Desired percentages used to set cmap control points.
+
+    Returns
+    -------
+    ctrl_pts : array
+        Array of floats corresponding to values to use as cmap control points.
+    """
+
+    # Based on type of limits specified, get cmap control points
+    if limits == 'auto':
+        ctrl_pts = np.percentile(np.abs(stc_data), [97, 99.9])
+        ctrl_pts.insert(1, np.average(ctrl_pts))
+    elif isinstance(limits, tuple):
+        assert len(limits) == 3, '"limits" tuple must be length 3'
+        ctrl_pts = limits
+    elif isinstance(limits, dict):
+        # Get appropriate key for limits if it's a dict
+        limit_key = ['lims', 'pos_lims'][colormap == 'mne_analyze']
+        assert limit_key in limits.keys(), ('"lims" OR "pos_lims" must be '
+                                            'defined depending on colormap')
+        if limits['kind'] == 'percent':
+            ctrl_pts = np.percentile(np.abs(stc_data),
+                                     list(np.abs(limits[limit_key])))
+        elif limits['kind'] == 'value':
+            ctrl_pts = limits[limit_key]
+        else:
+            raise ValueError('If limits is a dict, limits[kind] must be '
+                             ' "value" or "percent"')
+    else:
+        raise ValueError('"limits" must be "auto", tuple, or dict')
+
+    return ctrl_pts
+
+
```

Comment:
I figured out how to avoid repeating code, but kept it in a helper function. IMO, it makes the `plot_source_estimate` function cleaner/easier to parse at the cost of a few extra lines. Lmk though if I should do away with the private helper function though.